### DISCUSSION TITLE
Normal text is being rendered as MathJax in comments

### DISCUSSION THREAD

**Discussion by kmk3** [community/community#17522] [Other Feature Feedback, Questions, & Ideas]
In issue and discussion comments, normal text between dollar signs is being
rendered by MathJax.  Example:

${HOME}/foo should be moved to ${HOME}/bar.

That is rendered as:

`${HOME}/foo should be moved to $`{HOME}/bar.

Where the text in the code span becomes visually mangled, it cannot be selected
and right-clicking it opens a weird macOS-looking JavaScript-based context menu
that contains an "About MathJax" item.

Even though the text is not inside a code block or anything of the sort.

Environment: Firefox 100 on Linux.

Relates to #2816.

**Comment by kmk3** [comment]
Note that as of writing, this behavior is not documented anywhere on the
CommonMark spec, nor on the GitHub Flavored Markdown (GFM) spec:

* <https://spec.commonmark.org/0.30/> - Version 0.30 (2021-06-19)
* <https://github.github.com/gfm/> - Version 0.29-gfm (2019-04-06)

**Reply by rusty-snake** [reply]
https://docs.github.com/en/get-started/writing-on-github/working-with-advanced-formatting/writing-mathematical-expressions

**Comment by kmk3** [comment]
To use another programming-related resource for comparison, while multiple
sites on the Stack Exchange network enable the rendering of mathematical
expressions, it is not enabled on the more programming-focused sites, such as
on StackOverflow.  Additionally, some of those sites that enable the math
rendering use different delimiters.  From ["Which Stack Exchange sites use
MathJax?"][1]:

> These sites (and their respective Meta sites) support MathJax:
> 
> * [...]
> * Biology (with mhchem extension)
> * Blender
> * Chemistry (with mhchem extension)
> * Code Golf (with \\$ delimiters)
> * Code Review (with \\$ delimiters)
> * [...]
> * Earth Science (with mhchem extension)
> * Economics
> * Electrical Engineering (with \\$ delimiters)
> * Engineering
> * Game Development (with \\$ delimiters)
> * [...]
> * Matter Modeling (with mhchem extension)
> * [...]
> * Role-playing Games (with \\$ delimiters)

[1]: https://meta.stackexchange.com/questions/216606/which-stack-exchange-sites-use-mathjax/216607#216607

**Comment by PythonCoderAS** [comment]
Honestly I think there should at least be a per-repo/maybe even per-file way to turn this feature on or off.

**Reply by zakaryarashak** [reply]
17522

**Comment by rusty-snake** [comment]
https://github.blog/changelog/2022-05-19-render-mathematical-expressions-in-markdown/

It's a feature :wink:

**Comment by andrensairr** [comment]
I discovered GitHub's new feature quite by accident having made the simple but fatal mistake of using two $ signs in an issue comment. I'm all for new features, and I understand there are many who have long desired it, but I'm sure this won't be helpful for everybody and without a better way to explicitly markup math expressions, the approach taken runs the risk of breaking a heap of historical comments, not to mention confusing users of simple cases like myself.

As a work-around the dollar signs can be escaped by wrapping in `<span>` tags, a feature designed to avoid confusing the renderer when writing a $ signs inline with math expressions, but I think this is a heavy price to pay to opt out of the feature. Using the backslash (_a la_ `\$`) might have been a good choice, but alas this only works _within_ a math expression.

**Reply by joshjohanning** [reply]
I agree, if you have a two dollar signs in a line, it (sometimes) renders everything in between as math. 

Seems to be for example if you put a period right after a dollar sign and already have a dollar sign in the sentence. See: 

This renders incorrect:

Hi, I would like to buy your apples but they are $1, I only have $.75, can I buy 1?

```
Hi, I would like to buy your apples but they are $1, I only have $.75, can I buy 1?
```

This renders correctly:

Hi, I would like to buy your apples but they are $1, I only have $0.75, can I buy 1?

```
Hi, I would like to buy your apples but they are $1, I only have $0.75, can I buy 1?
```

**Comment by bence-toth** [comment]
According to [the documentation](https://docs.github.com/en/get-started/writing-on-github/working-with-advanced-formatting/writing-mathematical-expressions#writing-dollar-signs-in-line-with-and-within-mathematical-expressions) it is now recommended to use `<span>$</span>` to opt out of MathJax formatting on lines with multiple dollar signs:

E.g.:
```
<span>$</span>{HOME}/foo should be moved to ${HOME}/bar.
```

Will be rendered as:

<span>$</span>{HOME}/foo should be moved to ${HOME}/bar.

**Reply by rusty-snake** [reply]
Shorter alternative:

```markdown
&#36;{HOME}/foo
```

&#36;{HOME}/foo

**Comment by rusty-snake** [comment]
New alternative syntax (which doesn't seem to work):

https://github.blog/changelog/2022-06-28-fenced-block-syntax-for-mathematical-expressions

````
```math
\sqrt{5}
```
````

```math
\sqrt{5}
```
